src/solver/glu/backward_glu.cpp

```diff
@@ -46,14 +46,15 @@ namespace glu {
 bool GLUBackward::IsApplicable(const ExecutionContext& context,
                                const miopen::glu::ProblemDescription& problem) const
 {
-    std::ignore      = context;
-    auto input_numel = problem.GetInputDesc().GetElementSize();
+    std::ignore                  = context;
+    auto input_numel             = problem.GetInputDesc().GetElementSize();
+    size_t input_numel_threshold = 800000;
```

Comment:
Why is this limitation here? I don't see any technical obstacles to it. We always do benchmarking to choose the best algorithm for other primitives, but for this primitive there is only one algorithm. If there were more, and some other algorithm had better performance for these values, then this limitation would make sense.

Reply:
I have moved this limitation into `IsImprovementOverROCm()`, as it applies to cases where MIOpen performs better than ROCm.

Reply:
> Why is this limitation here? I don't see any technical obstacles to it. We always do benchmarking to choose the best algorithm for other primitives, but for this primitive there is only one algorithm. If there were more, and some other algorithm had better performance for these values, then this limitation would make sense.

@averinevg  We applied the limitation here because the algorithm only worked better comparing to ROCm/pytorch performance with this limitation. `isImprovementOverROCm` means the limitation for the algorithm to be better than ROCm/pytorch. For example please check [this PR](https://github.com/ROCm/MIOpen/pull/2583)   by our team did previously, argmax can only run with dim != last dim (we applied it because in case of last dim the algo's performance is worse than ROCm/pytorch), **there's no other algorithms that can work with dim = last dim**. Can you please clarify for our team that the algorithm we implemented should run without any limitations even if its performance is worse than ROCm/pytorch in some cases, or it should only apply to cases that is better than ROCm/pytorch?